rotkehlchen/chain/evm/decoding/odos/v2/decoder.py

```diff
@@ -0,0 +1,228 @@
+            token_addresses: list['ChecksumEvmAddress'],
+            token_amounts: list[int],
+    ) -> dict[str, FVal]:
+        """Returns the resolved evm tokens or native currency and their amounts"""
+        resolved_result: dict[str, FVal] = {}
+        for token_address, token_amount in zip(token_addresses, token_amounts, strict=True):
+            if token_address == ZERO_ADDRESS:
+                asset, decimals = (self.native_currency, self.native_currency_decimals)
+            else:
+                asset, decimals = ((token := get_or_create_evm_token(
+                    userdb=self.base.database,
+                    evm_inquirer=self.base.evm_inquirer,
+                    evm_address=token_address,
+                    chain_id=self.base.evm_inquirer.chain_id,
+                    token_kind=EvmTokenKind.ERC20,
+                )), token.get_decimals())
+
+            resolved_result[asset.identifier] = FVal(token_amount) / FVal(10 ** decimals)
+
+        return resolved_result
+
+    def _decode_swap(self, context: 'DecoderContext') -> 'DecodingOutput':
+        """Decodes swaps done using an Odos v1 router"""
+        if context.tx_log.topics[0] == b'\x82>\xaf\x01\x00-sS\xfb\xca\xdb.\xa30\\\xc4o\xa3]y\x9c\xb0\x91HF\xd1\x85\xac\x06\xf8\xad\x05':  # swapCompact()  # noqa: E501
+            decoded_data: tuple[ChecksumEvmAddress, list[int], list[ChecksumEvmAddress], list[int], list[ChecksumEvmAddress]] = (  # noqa: E501
+                hex_or_bytes_to_address(context.tx_log.data[0:32]),  # sender
+                [hex_or_bytes_to_int(context.tx_log.data[32:64])],  # input amount
+                [hex_or_bytes_to_address(context.tx_log.data[64:96])],  # input token address
+                [hex_or_bytes_to_int(context.tx_log.data[96:128])],  # output amount
+                [hex_or_bytes_to_address(context.tx_log.data[128:160])],  # output token address
+            )
+        elif context.tx_log.topics[0] == b'}\x7f\xb05\x18%:\xe0\x19\x13Sf(\xb7\x8dm\x82\xe6>\x19\xb9C\xaa\xb5\xf4\x94\x83V\x02\x12Y\xbe':  # swapMultiCompact()  # noqa: E501
+            try:
+                _, decoded_data = decode_event_data_abi_str(context.tx_log, SWAPMULTI_EVENT_ABI)  # type: ignore[assignment]  # types are known from the ABI
+            except DeserializationError as e:
+                log.error(
+                    f'Failed to deserialize Odos event {context.tx_log=} at '
+                    f'{context.transaction.tx_hash.hex()} due to {e}',
+                )
+                return DEFAULT_DECODING_OUTPUT
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        if not self.base.is_tracked(decoded_data[0]):
+            return DEFAULT_DECODING_OUTPUT
+
+        input_tokens = self._resolve_tokens_data(token_amounts=decoded_data[1], token_addresses=decoded_data[2])  # noqa: E501
+        output_tokens = self._resolve_tokens_data(token_amounts=decoded_data[3], token_addresses=decoded_data[4])  # noqa: E501
```

Comment:
is it always 4? CAn you maybe document what are those? I don't expect to have 2 tokens here